doc/website/blog/2021-08-27-visualizing-your-data-graph-using-entviz.md

```diff
@@ -0,0 +1,162 @@
+authorURL: "https://github.com/hedwigz"
+authorImageURL: "https://avatars.githubusercontent.com/u/8277210?v=4"
+authorTwitter: itsamitush
+---
+
+Joining an existing project with a large codebase can be a daunting task. Understanding the data model of an application is key for developers to start working on an existing project. One of the common tools to help developers grasp an application's data model is an ER (Entity Relation) diagram. ER diagrams provide a visual representation of your data model and detail each field of the entities. Many tools can help create them. For example, Jetbrains DataGrip can generate an ER diagram by connecting to and inspecting an existing database:
+
+![Datagrip ER diagram](https://link/to/image)
+
+
+[Ent](https://entgo.io/docs/getting-started/), a simple, yet powerful entity framework for Go, was originally developed inside Facebook specifically for dealing with projects with large and complex data models.
+This is why Ent uses code generation - it gives type-safety and code-completion out-of-the-box which helps explain the data model and improves dev velocity.
+On top of all of this, wouldn’t it be great to automatically generate ER diagrams that maintain a high-level view of the data model in a visually appealing representation? (I mean, who doesn’t love visualizations?) 
+
+### Introducing entviz
+[entviz](https://github.com/hedwigz/entviz) is an ent extension that automatically generates a static HTML page that visualizes your data graph.
+![Entviz example output](https://link/to/image)
+Most ER diagram generation tools need to connect to your database and introspect it, which makes it harder to maintain an up-to-date diagram of the DB. Since entviz integrates directly to your Ent schema, it does not need to connect to your database, and it automatically generates fresh visualization every time you modify your schema.
+
+If you want to know more about ent how entviz was implemented, checkout the [implementation section](#implementation).
+
+  
+### Usage
+First, let’s add entviz extension to our entc.go file:
+```bash
+go get github.com/hedwigz/entviz
+```
+:::info
+If you are not familiar with `entc` you're welcome to read [entc documentation](https://entgo.io/docs/code-gen#use-entc-as-a-package) to learn more about it.
+:::
+```go title="ent/entc.go"
+err := entc.Generate("./schema", &gen.Config{}, entc.Extensions(entviz.Extension{}))
+ if err != nil {
+   log.Fatalf("running ent codegen: %v", err)
+ }
+```
+Let’s say we have a simple schema with a user entity, and some fields:
+```go title="ent/schema/user.go"
+// Fields of the User.
+func (User) Fields() []ent.Field {
+ return []ent.Field{
+   field.String("name"),
+   field.String("email"),
+   field.Time("created").
+     Default(time.Now),
+ }
+}
+```
+Now, entviz will automatically generate a visualization of our graph everytime we run: 
+```bash
+go generate ./...
+```
+You should now see a new file called `schema-viz.html` in your ent directory
```

Comment:
Great. This cause/effect makes the read much more pleasant, thanks for adding this